Handle cases where a route's prefix is a nested URL

SimonKaluza opened the pull request:
Nested prefixes (any prefix containing a '/' character) prior to this commit would incorrectly get tagged as the first segment of the URL prefix.

E.g. a route like "api/beta/v1/patients" would incorrectly get tagged as "api" instead of "patients"

LeFnord commented:
mmh for me the changes are reasonable, but please adept the specs and/or add one which documents the behaviour

SimonKaluza commented:
Added a RSpec example with tags that breaks without the adjustments made to endpoint.rb to support nested prefix and rebased off of the current master (hence the force-pushes)

LeFnord commented:
thanks @SimonKaluza
